src/input/user_input.rs

```diff
@@ -807,6 +808,10 @@ where
             }
             Message::ChooseFile(my_index_clone.clone(), result, input_data_clone.clone())
         });
+        let extensions = extensions
+            .as_ref()
+            .unwrap_or(&vec![String::new()])
+            .join(",");
```

Comment:
I understand current logic as follows:

- If `extensions` is `None`, no extensions are allowed.
- If `extensions` is `Some`, only the extensions listed in the `Vec<String>` are allowed. This means that if the vector is empty, no extensions are allowed.

There is no functional difference between `None` and `Some(vec![])`. This suggests that the use of `Option` is unnecessary. 

Therefore, I suggest we either change `extensions` and `allowed_extensions` to just `Vec<String>`, or keep it as `Option<Vec<String>>` but change the behavior like below:

- If `extensions` is `None`, all extensions are allowed.

I believe one of these two directions would be more appropriate.

Reply:
Thank you for the feedback! I’ve implemented Option 1 by replacing `Option<Vec<String>>` with `Vec<String>`, simplifying the logic. Let me know if there’s anything else you’d like me to adjust!